compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -33,28 +33,27 @@ pub fn find_native_static_library(
     search_paths: &[PathBuf],
     sess: &Session,
 ) -> PathBuf {
-    let verbatim = verbatim.unwrap_or(false);
-    // On Windows, static libraries sometimes show up as libfoo.a and other
-    // times show up as foo.lib
-    let oslibname = if verbatim {
-        name.to_string()
+    let formats = if verbatim.unwrap_or(false) {
+        vec![("".to_string(), "".to_string())]
     } else {
-        format!("{}{}{}", sess.target.staticlib_prefix, name, sess.target.staticlib_suffix)
+        let os =
+            (sess.target.staticlib_prefix.to_string(), sess.target.staticlib_suffix.to_string());
+        // On Windows, static libraries sometimes show up as libfoo.a and other
+        // times show up as foo.lib
+        let unix = ("lib".to_string(), ".a".to_string());
+        if os == unix { vec![os] } else { vec![os, unix] }
     };
-    let unixlibname = format!("lib{}.a", name);
 
-    for path in search_paths {
-        let test = path.join(&oslibname);
-        if test.exists() {
-            return test;
-        }
-        if oslibname != unixlibname {
-            let test = path.join(&unixlibname);
+    for format in formats {
+        for path in search_paths {
```

Comment:
Previously `search_paths` were traversed in the outer loop, and prefixes/suffixes in an inner "loop".
Now the order is swapped, which is a potentially observable change.